core/storage/feedback/gae_models.py

```diff
@@ -420,6 +421,400 @@ def get_multi(cls, user_id, thread_ids):
         return super(FeedbackThreadUserModel, cls).get_multi(instance_ids)
 
 
+class GeneralFeedbackThreadModel(base_models.BaseModel):
+    """Threads for each entity.
+
+    The id of instances of this class has the form
+        [ENTITY_TYPE].[ENTITY_ID].[GENERATED_STRING]
+    """
+    # The type of entity the thread is linked to.
+    entity_type = ndb.StringProperty(required=False, indexed=True)
+    # The ID of the entity the thread is linked to.
+    entity_id = ndb.StringProperty(required=False, indexed=True)
+    # ID of the user who started the thread. This may be None if the feedback
+    # was given anonymously by a learner.
+    original_author_id = ndb.StringProperty(indexed=True)
+    # Latest status of the thread.
+    status = ndb.StringProperty(
+        default=STATUS_CHOICES_OPEN,
+        choices=STATUS_CHOICES,
+        required=True,
+        indexed=True,
+    )
+    # Latest subject of the thread.
+    subject = ndb.StringProperty(indexed=False)
+    # Summary text of the thread.
+    summary = ndb.TextProperty(indexed=False)
+    # Specifies whether this thread has a related learner suggestion.
+    has_suggestion = ndb.BooleanProperty(indexed=True, default=False)
+    # The number of messages in the thread.
+    message_count = ndb.IntegerProperty(indexed=True)
```

Comment:
Presumably also required.

Reply:
I am not sure if that will work fine. I saw this TODO, not sure if it resolved. (If it is, we can make this required I think)
https://github.com/oppia/oppia/blob/41fdbb327ab40b64824f01d7c18e12a2158b48dd/core/domain/feedback_services.py#L558-L565

Reply:
Oh, OK. Not sure either. @Arunabh98 do you remember?

Otherwise, file issue to resolve later.

Reply:
Ok. Will file an issue. (Later we could run a one off job that checks for the presence of message_count, and then resolve this TODO).